plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java

```diff
@@ -183,7 +183,6 @@ public void allRequiredSinksFinished()
                 return;
             }
             verify(noMoreSinks, "noMoreSinks is expected to be set");
-            verify(finishedSinks.keySet().containsAll(allSinks), "all sinks are expected to be finished");
```

Comment:
do we need to finish/close the sinks that weren't finished yet?
can there be a resource leak we if we do not so?

Reply:
I also though about it. Theoretically yeah - but not the case for exchange implementation we have right now. So I would not extend SPI just yet - we can add the method if we see it is needed for some upcoming exchange impl.

Reply:
> can there be a resource leak we if we do not so?

maybe not a leak, but memory could be stalled for a longer time then needed. I would close them if we can